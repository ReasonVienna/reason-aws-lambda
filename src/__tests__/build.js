// @flow

jest.setMock(
  "../buildArtifact",
  jest.fn(({ zipFile }) => Promise.resolve(zipFile))
);

const path = require("path");
const build = require("../build");
const buildArtifact = _mock(require("../buildArtifact"));

jest.mock("fs");
jest.mock("../buildArtifact");

function _mock(fn: any): JestMockFn {
  return fn;
}

it("should create test.zip assuming cwd to be a ReasonProject", async () => {
  buildArtifact.mockClear();

  const logger = {
    log: jest.fn(),
    error: jest.fn()
  };

  const result = await build({
    cwd: "/some/cwd",
    functions: {
      test: {
        path: "src/Test.re",
        handler: "test"
      }
    },
    buildDir: path.join("/some/cwd", "_build"),
    logger
  });

  const {
    reFile,
    zipFile,
    context
  } = buildArtifact.mock.calls[0][0];

  expect(reFile).toEqual("src/Test.re");
  expect(zipFile).toEqual("/some/cwd/_build/test.zip");
  expect(context).toEqual("/some/cwd");

  expect(logger.log.mock.calls).toMatchSnapshot();
  expect(result).toEqual(["/some/cwd/_build/test.zip"]);
});

it("should create test.zip with a different context path", async () => {
  buildArtifact.mockClear();

  const logger = {
    log: jest.fn(),
    error: jest.fn()
  };

  const result = await build({
    context: "/reason",
    cwd: "/some/cwd",
    functions: {
      test: {
        path: "src/test.re",
        handler: "test"
      }
    },
    logger,
    buildDir: path.join("/some/cwd", "_build")
  });

  expect(logger.log.mock.calls).toMatchSnapshot();
  expect(result).toEqual(["/some/cwd/_build/test.zip"]);
});
