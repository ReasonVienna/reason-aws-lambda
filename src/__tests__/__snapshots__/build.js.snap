// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create test.zip assuming cwd to be a ReasonProject 1`] = `
Array [
  Array [
    "Creating build directory '/some/cwd/_build'...",
  ],
  Array [
    "Created zip /some/cwd/_build/test.zip",
  ],
]
`;

exports[`should create test.zip with a different context path 1`] = `
Array [
  Array [
    "Creating build directory '/some/cwd/_build'...",
  ],
  Array [
    "Created zip /some/cwd/_build/test.zip",
  ],
]
`;
